### PR TITLE
Delete files/functions/tests we suspect will be no longer necessary after IVADOMED transition

### PR DESCRIPTION
**Pull request by mathieuboudreau**
See: https://docs.google.com/document/d/1Ipd6_nK6I6CKlDI_KpkLXH_jY2C547W_ootalfKnFlU/edit?usp=sharing

## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [ ] I've added relevant tests for my contribution
- [ ] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer
- [ ] I've consulted [ADS's internal developer documentation](https://github.com/neuropoly/axondeepseg/wiki/Guidelines-for-contribution) to ensure my contribution is in line with any relevant design decisions

## Description
Delete files/functions/tests we suspect will be no longer necessary after IVADOMED transition

## Linked issues
#523

### CODE REVIEW

File: AxonDeepSeg/visualization/visualize.py [hunk 1]
```diff
@@ -1,265 +0,0 @@
...
-        ax2.set_ylabel("Loss")
-        return fig
-
-    evolution = retrieve_training_data(path_model)
-    fig = _create_figure_helper(evolution)
-
-    return fig
-
-
-def visualize_segmentation(path):
```
Comment (mariehbourget) [inline AxonDeepSeg/visualization/visualize.py:74]: I think I would keep the `visualize_segmentation` function in the code for now and only remove the other functions related to training.

Similar to the functions of the other files in the visualization folder, it look like a useful tool but is not used at the moment. I would keep it for now, but we should re-assess its usage later in the integration process with ivadomed.

**Review by mariehbourget — COMMENTED** [review]
Thank you @mathieuboudreau and @vasudev-sharma for the detailed comparison table and first PR. I went through the table again and add some comments following our meeting.

As for the PR, there is only the function `visualize_segmentation` from `visualize.py` that I would keep for now, everything else looks good to me and match the table.

File: AxonDeepSeg/visualization/visualize.py (hunk 1 above)
Comment (mathieuboudreau) [reply AxonDeepSeg/visualization/visualize.py:74]: Done in be539d3

**Comment by mathieuboudreau** [comment]
@mariehbourget @vasudev-sharma I think this PR is ready for merging, all that's needed is a quick look at all the files deletes to ensure that it agrees with the list that we curated in PR #523

**Review by mariehbourget — APPROVED** [review]
LGTM! I checked the new changes. I did not compare the entire table again because the last time, the only difference was `visualize_segmentation` which was fixed.
Thank you @mathieuboudreau!

**Pull request merged**